### PR TITLE
Add examples section to documentation with initial rough-draft examples

### LINKED ISSUE #55: Restructure docs to have a Tutorials and Examples section (or two sections, for those things)
I think the current structure of the docs is not conducive to learning and easy access.

We should have a tutorial as part of the getting started component, and a suite of examples as an appendix.
It doesn't make a lot of sense for the `GlobusAuthorizer` docs to have a large example section embedded in the middle of it, and the structured code documentation generally seems mixed up in the middle of the high-level meta.

This is a clear separation that I'd like for us to enforce -- I think it will clean things up significantly.

### PR DESCRIPTION
**Pull request by sirosen**
@jaswilli, @corpulentcoffee I'd like to push this through as soon as either of you can do a cursory review for glaring faults.
It's not the most refined documentation, but it resolves #55, the only remaining issue I had slated for `0.5.0`.
I'm not sure how much time anyone has for a detailed review.

There are currently four examples, with no particular order or priority:
 - `authorization` covers basic usage of `GlobusAuthorizer`s
 - `native_app` covers a Native App login with copy-paste
 - `client_credentials` covers a simple Client Credentials grant
 - `three_legged_oauth` is a more extensive example of Authorization Code login using Flask
   (not as featureful as the Sample Data Portal; skips out on using `id_token`)

### CODE REVIEW

**Comment by corpulentcoffee** [comment]
@aaschaer Can you take a review at these new docs (e.g. check for readability but also try the code examples) from Stephen?

File: docs-source/examples/three_legged_oauth.rst [hunk 1]
```diff
@@ -0,0 +1,191 @@
...
+    $ pip install Flask==0.11.1 globus-sdk
+    ...
+
+You'll also want a shared function for loading the SDK ``AuthClient`` which
+represents your application, as you'll need it in a couple of places. Create
+it, along with the defintiion for your Flask app, in ``example_app.py``:
+
+.. code-block:: python
+
+    from flask import Flask, url_for, session
```
Comment (aaschaer) [inline docs-source/examples/three_legged_oauth.rst]: A few more imports are needed from flask to run the example
`from flask import Flask, url_for, session, redirect, request`

File: docs-source/examples/three_legged_oauth.rst [hunk 2]
```diff
@@ -0,0 +1,191 @@
...
+    import globus_sdk
+
+    app = Flask(__name__)
+    app.config.from_pyfile('example_app.conf')
+
+    def load_app_client():
+        return globus_sdk.ConfidentialAppAuthClient(
+            app.config['APP_CLIENT_ID'], app.config['APP_CLIENT_SECRET'])
+
```
Comment (aaschaer) [inline docs-source/examples/three_legged_oauth.rst]: I'm not entirely sure this is the best place, but the example code needs to run the app somewhere.
`if __name__ == "__main__":`
    `app.run()`

File: docs-source/examples/three_legged_oauth.rst [hunk 3]
```diff
@@ -0,0 +1,191 @@
...
+Login sends a user to Globus Auth to get credentials, and then may act on the
+user's behalf.
+Logout invalidates server-side credentials, so that the application may no
+longer take actions for the user, and the client-side session,
+allowing for a fresh login if desired.
+
+Register an App
+~~~~~~~~~~~~~~~
+
+Go to developers.globus.org and register a new application.
```
Comment (aaschaer) [inline docs-source/examples/three_legged_oauth.rst]: This should probably be a hyper link. Also I was wondering if all the examples should point here? Its much easier to find the registration form through this link, although there isn't an explanation of what Native vs non-Native apps are if we don't send them to the Auth docs.

File: docs-source/examples/client_credentials.rst [hunk 4]
```diff
@@ -0,0 +1,95 @@
...
+
+    import globus_sdk
+
+    # you must have a client ID
+    CLIENT_ID = '...'
+    # the secret, loaded from wherever you store it
+    CLIENT_SECRET = '...'
+
+    client = globus_sdk.ConfidentialAppAuthClient(CLIENT_ID, CLIENT_SECRET)
+    token_response = client.oauth2_client_credentials_tokens(auth_code)
```
Comment (aaschaer) [inline docs-source/examples/client_credentials.rst]: `auth_code` isn't defined or needed here, `token_response = client.oauth2_client_credentials_tokens()` works fine.

**Comment by sirosen** [comment]
Updated per @aaschaer's review, and made the "Get a Client" instructions uniform. They now link both to the Auth docs and to the developers page.

I think this is good to ship?

**Pull request merged**